src/Leaflet.Renderer.Canvas.Tile.js

```diff
@@ -43,17 +43,17 @@ L.Canvas.Tile = L.Canvas.extend({
 	},
 
 	_onClick: function (e) {
-		var point = this._map.mouseEventToLayerPoint(e).subtract(this.getOffset()), layer, clickedLayer;
+		var point = this._map.mouseEventToLayerPoint(e).subtract(this.getOffset()), layers = [], layer;
```

Comment:
Let's rename this `layers` array to `clickedLayers` or something like this since this is a list of layers you are have triggered the event and it might be confusing to have two `layers` list in the same scope (`this._layers` and `layers`) which are apparently the same thing